src/coap_debug.c

```diff
@@ -77,19 +77,24 @@ static const char *loglevels[] = {
 COAP_STATIC_INLINE size_t
 print_timestamp(char *s, size_t len, coap_tick_t t) {
   struct tm *tmp;
+  int lensofar;
   time_t now = coap_ticks_to_rt(t);
   tmp = localtime(&now);
-  return strftime(s, len, "%b %d %H:%M:%S", tmp);
+  strftime(s, len, "%b %d %H:%M:%S", tmp);
+  lensofar = strlen(s);
+  snprintf(&s[lensofar], len-lensofar, ".%03u",
```

Comment:
I think this occurrence of `snprintf` should be guarded with `#ifdef HAVE_SNPRINTF` as well for consistency (c.f. the alternative version of `print_timestamp` below).

Reply:
HAVE_SNPRINTF is only defined for the contiki build, so having the guard breaks things.

Reply:
Hm, ok. It seems that the rationale at some point was that platforms with `time.h` also have `snprintf`. Not really convincing but I agree that we should not introduce guards here that might break a lot of other stuff out there.